Update user guide for autocomplete and todo (#153)
AY2223S1-CS2103T-W15-1/tp: docs/UserGuide.md

Adds a todo to your SoConnect.

Format: `todo add d/DESCRIPTION date/DATE pr/PRIORITY [t/TAG]…​`

* `DATE` should be of the format dd-MM-yyyy (e.g. 24-03-2022).
* `PRIORITY` can only be `low`, `medium`, `high`.

> * The todo list will always be sorted by date from earliest to latest, then follow by decreasing priority.

Reviewer: ```suggestion
* The todo list will always be sorted by date from earliest to latest (For todos with the same date, they will be sorted in decreasing priority order).
```

Shows a filtered list of todos in your SoConnect.

Format: `todo show`, `todo show today`, `todo show date/DATE`, `todo show date/DATE to DATE`, `todo show t/TAG`, `todo show pr/Priority`

* `todo show`: Shows all todos.
* `todo show today`: Shows all todos with the date same as the current date.
* `todo show date/DATE`: show all todos with the specified date.
* `todo show date/DATE1 to DATE2`: shows all todos with the date from `DATE1` to `DATE2`.

> * `pr/PRIORITY`: Shows all todos with the specified priority.

Reviewer: ```suggestion
* `todo show pr/PRIORITY`: Shows all todos with the specified priority.
```

Author: Done!

> * `t/TAG`: Shows all todos with the specified tag.

Reviewer: ```suggestion
* `todo show t/TAG`: Shows all todos with the specified tag.
```

Author: Okay done!
